Apps/Examples/Examples/All Examples/SymbolClusteringExample.swift

```diff
@@ -0,0 +1,166 @@
+
+        view.addSubview(mapView)
+
+        // Add the source and style layers once the
+        mapView.mapboxMap.onNext(.mapLoaded) { _ in
+            self.addSymbolClusteringLayers()
+        }
+        
+        let tap = UITapGestureRecognizer(target: self, action: #selector(handleTap(gestureRecognizer:)))
+        mapView.addGestureRecognizer(tap)
+    }
+
+    func addSymbolClusteringLayers() {
+        let style = self.mapView.mapboxMap.style
+        // The image named `fire-station-11` is included in the app's Assets.xcassets bundle.
+        let image = UIImage(named: "fire-station-11")!.withRenderingMode(.alwaysTemplate)
+        // Set `sdf` to `true`. This allows the icon images to be recolored.
+        try! style.addImage(image, id: "fire-station-icon", sdf: true)
+
+        // Fire_Hydrants.geojson contains information about fire hydrants in the District of Columbia.
+        // It was downloaded on 6/10/21 from https://opendata.dc.gov/datasets/DCGIS::fire-hydrants/about
+        guard let url = Bundle.main.url(forResource: "Fire_Hydrants", withExtension: "geojson") else {
+            return
+        }
+
+        // Create a GeoJSONSource using the previously specified URL.
+        var source = GeoJSONSource()
+        source.data = .url(url)
+
+        // Enable clustering for this source.
+        source.cluster = true
+        source.clusterRadius = 75
+        let sourceID = "fire-hydrant-source"
+
+        var clusteredLayer = createClusteredLayer()
+        clusteredLayer.source = sourceID
+
+        var unclusteredLayer = createUnclusteredLayer()
+        unclusteredLayer.source = sourceID
+
+        // Add the source and two layers to the map.
+        try! style.addSource(source, id: sourceID)
+        try! style.addLayer(clusteredLayer)
+        try! style.addLayer(unclusteredLayer, layerPosition: .below(clusteredLayer.id))
+
+        // This is used for internal testing purposes only and can be excluded
+        // from your implementation.
+        finish()
+    }
+
+    func createClusteredLayer() -> SymbolLayer {
+        // Create a symbol layer to represent the clustered points.
+        var clusteredLayer = SymbolLayer(id: "clustered-fire-hydrant-layer")
+        clusteredLayer.filter = Exp(.has) { "point_count" }
+
+        clusteredLayer.iconImage = .constant(.name("fire-station-icon"))
+
+        // Set the color of the icons based on the number of points within
+        // a given cluster. The first value is a default value.
+        clusteredLayer.iconColor = .expression(Exp(.step) {
+            Exp(.get) { "point_count" }
+            UIColor(red: 0.12, green: 0.90, blue: 0.57, alpha: 1.00)
+            50
+            UIColor(red: 0.12, green: 0.53, blue: 0.90, alpha: 1.00)
+            100
+            UIColor(red: 0.85, green: 0.11, blue: 0.38, alpha: 1.00)
+        })
+
+        // Add an outline to the icons.
+        clusteredLayer.iconHaloColor = .constant(.init(color: .black))
```

Comment:
Does using type inference to say `.init` make this code more or less clear? (I'd argue less)